### ISSUE TITLE
Layers/depth seem non-optimal

### ISSUE THREAD

**Issue by henkjan-sneller** [brianpursley/sorting-network#2]
While plotting a machine generated network I noticed that the order of nodes matters (for the layer assignment) while it should not AFAICT.

For example 2:5,0:1,4:5,0:3 yields:
![exampleNet](https://user-images.githubusercontent.com/83118252/220623434-4bc85329-82b2-466f-9304-c083e1964508.svg)

While 2:5,4:5,0:1,0:3 yields:
![exampleNet](https://user-images.githubusercontent.com/83118252/220623248-0a959ea5-4f99-49f7-a956-072dbf3a98c4.svg)
and I expected it to be the same as the previous one since the layers and dephts did not change.

Q: is this a bug or a feature?

**Comment by brianpursley** [comment]
Yes, this is a bug.

It needs some code to "collapse" the network so that there are no unnecessary gaps at any given depth.

**Comment by brianpursley** [comment]
@henkjan-sneller Thanks for reporting this.  It should be fixed by a change I just made.

**Comment by HJLebbink** [comment]
Thanx!

**Comment by brianpursley** [comment]
The optimal depth grouping has been fixed, but it seems within a single depth, the comparators are still not organized as well as they could be.

![Screenshot_20240506-184655](https://github.com/brianpursley/sorting-network/assets/4977542/f5b9346f-032b-4af7-bfcc-6c3c93152d9d)

So maybe there is still some work that can be done to improve this.

**Comment by brianpursley** [comment]
Here is a simplified reproduction of the layout problem that remains, where comparators are not positioned in the most optimal way within a depth group.

```
echo "0:2,1:5,3:4" | ./sortingnetwork.py --svg | rsvg-convert > layout.png
```

![image](https://github.com/user-attachments/assets/8650de65-a49b-4b96-879c-a6d5d220cf94)

Expected output should be:

![image](https://github.com/user-attachments/assets/fdcffb67-8da5-4f0c-8280-f6bfebda62bb)

I'm working on a fix for this...

**Comment by brianpursley** [comment]
Fixed by #7
